### DISCUSSION TITLE
DIY Loop Failed Error code 2

### DISCUSSION THREAD

**Discussion by LisanneLoop** [fastlane/fastlane#22214] [Q&A]
Hi There,
I use DIY LOOP. Now a few days ago my app expired, i have to rebuild every 3 months.
I got a error: 
Unable to create a valid authorization token for the App Store Connect API. Verify that the FASTLANE_ISSUER_ID, FASTLANE_KEY_ID, and FASTLANE_KEY secrets are set correctly and try again.

Now i checked the forum and i saw it has to do something with Apple Developer.
So i figured out my credit card was expired. I changed it to an up to date credit card. But i'm still getting the same error.

Does anyone have an idea what's going wrong?

Thanks!!

Grtz
Lisanne

**Comment by Krazoa584** [comment]
I have the same problem and unfortunatelly no answer :(

**Comment by Krazoa584** [comment]
Hey, I think a  have a solution. For me it worked when I updated with the help of this article (from chapter "Renew Certificate") https://loopkit.github.io/loopdocs/browser/bb-update/#accept-agreements

**Reply by MightyThorn** [reply]
This worked for me.  Thanks for posting!  There was a new agreement I hadn't yet agreed to

**Comment by AnneN87** [comment]
Hi,

Same problem here, unfortunately..

" Error: Unable to create a valid authorization token for the App Store Connect API. Verify that the FASTLANE_ISSUER_ID, FASTLANE_KEY_ID, and FASTLANE_KEY secrets are set correctly and try again.
Error: Process completed with exit code 2. "

I triple checked the Payment method (CC), Apple Developer Licence Agreements are agreed, I twice remade the Apple Store Connect (FastLane) API with the 2 keys, updated and double checked the 4 Apple Identifiers. So I think all the inputs should be fine, but somehow I still get the error message.

My Apple Certificate (still valid) is Development Type. Is that maybe it?

Or has anybody any idea what I need to do to fix this?

Many thanks!!
Anne

**Comment by HenrikHL** [comment]
This also stopped working for me yesterday... 

The automated builds have been running for months...

**Reply by HenrikHL** [reply]
Logging into [https://developer.apple.com/account](https://developer.apple.com/account) and accepting the new terms solved the problem.
**Note:** Remember to wait a couple of minutes before rerunning the `Build Loop` workflow again

**Reply by HenrikHL** [reply]
After renewal of my Apple Developer License my Certificates would not create. Deleting all "Expired" profiles solved the problem for me.
<img width="678" alt="image" src="https://github.com/user-attachments/assets/9f899a7d-f668-469c-ac6a-f5c1de4026d0" />

**Comment by dvayanu** [comment]
Have the same issues, everything agreed and confirmed and what not, and when I build something (else) with xcode, I can distribute it alright, so the account is ok. Generated new set of keys for fastlane, the error still there. Anyone from above has been able to resolve the problem?
As for developer programm and agreements I confirmed them all at least 30 hours ago, still no build.

**Reply by ahawking** [reply]
Having the same issue

**Reply by dvayanu** [reply]
I posted to the loopandlearn group on facebook and they were able to help me, turned out my distribution certificate expired
https://loopkit.github.io/loopdocs/browser/bb-update/?h=renew&fbclid=IwY2xjawH_RcxleHRuA2FlbQIxMAABHUFLnwgT9Wlr2qu0i7HFmiBlBuN-N0e9dD7RwX5qBRvJyHsVIk5AifWNoA_aem_dSnFstyKR2x5AXXlME0VIw#manual-steps-to-renew-your-distribution-certificate

The group is very helpful: https://www.facebook.com/groups/2161109864145506
